### ISSUE TITLE
how to install it?

### ISSUE THREAD

**Issue by ksrvpvl** [bitfocus/companion-module-crestron-digitalmatrix#1]
It's great that there is such a module, but how to install it?

**Comment by krocheck** [comment]
@llamafilm did you ever get this working?  I'm not seeing a request history in Slack asking for this module to be checked and added to the betas.

**Comment by llamafilm** [comment]
Yes it works for making a route. I haven’t had any trouble with it.
I’m not familiar with the process of adding it to beta. I just keep the folder inside module-local-dev.

**Comment by krocheck** [comment]
Ok.  One of the issues I see with releasing this as-is is that it doesn't support a password.  TELNET and CTP can have a password prompt if set, and I know this because all IP devices we deployed when I worked at a university required CTP to be password protected.  If I were a betting man and we put this out, adding password support will be one of the first tickets.  Any chance you can add a config field for that and test it?

**Comment by llamafilm** [comment]
I've just added a password option in https://github.com/bitfocus/companion-module-crestron-digitalmatrix/commit/a2528f3c1544c766cd26cfef875fc143782f01d2.  It works fine on my system, but it would be good to have some more testers.  I'm not sure if the telnet syntax is always the same across different models.

**Comment by ksrvpvl** [comment]
great, I have a matrix of 16. how should I try?  how to install dev?

**Comment by llamafilm** [comment]
Download this repo and place the companion-module-crestron-digitalmatrix folder inside module-local-dev folder (inside Companion folder) then restart Companion.

**Comment by llamafilm** [comment]
I still need to add support for bigger matrices. Currently you’ll only be able to route the first 8 sources and destinations.

**Comment by ksrvpvl** [comment]
please add instructions on how to install this module! I copied the folder along the way C:\Program Files\Companion\resources\module-legacy\manifests\companion-module-crestron-digitalmatrix-main
how do I add a module?

**Comment by llamafilm** [comment]
Running modules in developer mode is for advanced users only.  Instructions are here: https://github.com/bitfocus/companion/wiki/Developing-on-Windows-10

This is supposed to get easier with version 3, but I haven't had time to rewrite this module for v3 yet.

**Comment by llamafilm** [comment]
Ok I took some time to get this updated for v3.  So now the installation is much easier.  From the red launcher window, click the settings gear and set the developer modules path to any folder you like.  Then download the `v3` branch and place the `companion-module-crestron-digitalmatrix` folder inside there.

**Comment by ksrvpvl** [comment]
it doesn't work, I did everything according to your words.

**Comment by ksrvpvl** [comment]
it can't be installed, why is it here? or post how to install it...

**Comment by ksrvpvl** [comment]
even after adding, i get errors
`error: node:internal/modules/cjs/loader:1078

  throw err;

  ^

Error: Cannot find module '../../tcp'

Require stack:

- D:\bitfocus\companion-module-crestron-digitalmatrix-main\index.js

    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)

    at Module._load (node:internal/modules/cjs/loader:920:27)

    at Module.require (node:internal/modules/cjs/loader:1141:19)

    at require (node:internal/modules/cjs/helpers:110:18)

    at Object.<anonymous> (D:\bitfocus\companion-module-crestron-digitalmatrix-main\index.js:1:13)

    at Module._compile (node:internal/modules/cjs/loader:1254:14)

    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)

    at Module.load (node:internal/modules/cjs/loader:1117:32)

    at Module._load (node:internal/modules/cjs/loader:958:12)

    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {

  code: 'MODULE_NOT_FOUND',

  requireStack: [

    'D:\\bitfocus\\companion-module-crestron-digitalmatrix-main\\index.js'

  ]

}



Node.js v18.14.2
`

**Comment by Nahon-od** [comment]
Does anyone know how to get this module to work? Please share

**Comment by ksrvpvl** [comment]
[h07d0q](https://github.com/h07d0q), I have dm16x16, it doesn't work, just nothing happens

**Comment by h07d0q** [comment]
module will be added to companion by developer team to some of the next releases
